src/Functions/extractTimeZoneFromFunctionArguments.cpp

```diff
@@ -30,11 +30,11 @@ std::string extractTimeZoneNameFromColumn(const IColumn * column, const String &
 }
 
 
-std::string extractTimeZoneNameFromFunctionArguments(const ColumnsWithTypeAndName & arguments, size_t time_zone_arg_num, size_t datetime_arg_num, bool allow_nonconst_timezone_arguments)
+std::string extractTimeZoneNameFromFunctionArguments(const ColumnsWithTypeAndName & arguments, size_t time_zone_arg_num, size_t datetime_arg_num, [[maybe_unused]] bool allow_nonconst_timezone_arguments)
```

Comment:
Argument `allow_nonconst_timezone_arguments` is no longer used. Could you maybe just remove it (and adjust the callers)?